### ISSUE TITLE
XPath function for accessing FeatureTypes from Schema

### ISSUE THREAD

**Issue by jonherrmann** [interactive-instruments/etf-webapp#37]
Implement in etf-bsx-ae

**Comment by michellutz** [comment]
@jonherrmann, as discussed during the 2017.4 meeting on 2018-03-16, please elaborate a little bit on what needs to be done to implement this change.

**Comment by jonherrmann** [comment]
@michellutz updated the description.

**Comment by cportele** [comment]
@jonherrmann @michellutz 

I think one of the original ideas was to support a similar capability to `schema-element()` from XSLT, too. This would be useful when testing INSPIRE data that uses an extension. 

The current situation is that the ETSs use local element names to identify feature instances of a specific feature type. Ie. the ETSs will also identify and test features of feature types that include a few additional attributes in the namespace of an extension. These feature types will have the same local name as they implement the same feature concept (if the extension follows the conceptual model). However, proper specialisations of the INSPIRE feature type will typically receive different local names and cannot be detected as subtypes of an INSPIRE feature type by looking at the element name. Schema analysis is required. 

The approach described in the initial description does not seem to support this. Or has it been discussed in Ispra that the current support for extensions is sufficient?

**Comment by jonherrmann** [comment]
@cportele maybe the wording 'by parsing the associated schema' is not clear enough -'reading and analyzing' might have been better- but under the hood this functionality requires a similar capability to the schema-element() function.

So the result of `//schema-element(gml:AbstractFeature)`would be equal to `etf:feature-members( / )` . Would this be OK?

I will also add the schema-element() function to the numbered list in the description, so that it can be used for other types of tests.

**Comment by cportele** [comment]
Yes, `etf:feature-members( / )` is the same as `//schema-element(gml:AbstractFeature)`, but for the ETSs the typical expression needed would be something like `//schema-element(ps:ProtectedSite)` (where protected site is an example) - or even better something like `$features[etf:is-kind-of(., ps:ProtectedSite)]` or similar.

**Comment by jonherrmann** [comment]
Ah, that makes sense -sorry, I overlooked that part..
